djangoevents/domain.py

```diff
@@ -47,8 +71,43 @@ class BaseEntity(BaseAggregate):
 
     OBSOLETE! Interface kept for backward compatibility.
     """
+    @classmethod
+    def is_abstract_class(cls):
+        return super().is_abstract_class() or cls is BaseEntity
 
     @classmethod
     def mutate(cls, entity=None, event=None):
         warnings.warn("`BaseEntity` is depreciated. Please switch to: `BaseAggregate`", DeprecationWarning)
         return super().mutate(aggregate=entity, event=event)
+
+
+def _list_subclasses(cls):
+    """
+    Recursively lists all subclasses of `cls`.
+    """
+    return cls.__subclasses__() + [g for s in cls.__subclasses__() for g in _list_subclasses(s)]
+
+
+def _list_internal_classes(cls, base_class=None):
+    base_class = base_class or object
+
+    return [cls_attribute for cls_attribute in cls.__dict__.values()
+            if inspect.isclass(cls_attribute)
+            and issubclass(cls_attribute, base_class)]
+
+
+def list_concrete_aggregates():
```

Comment:
I'd either parametrise or rename, so that it's more clear. Still this is a library function, so somebody might want to use in a different situation :)